### ISSUE TITLE
Compiler test for GHC 8.0.2 takes too long

### ISSUE THREAD

**Issue by andreasabel** [agda/agda#4018]
Issue #3239 is back, the travis instance for 8.0.2 runs out of time due to the compiler-test (AllStdLib).  The fix then was an increase of the RTS parameter `-H` to 2G.  The maximum heap size `-M` is currently 3G.
 
See also #3792, which was fixed by optimizing the GHC backend.

Seems like from build 6083 on https://travis-ci.org/agda/agda/builds/573485300  the failure is persistent.

**Comment by L-TChen** [comment]
I'd suggest to use Stack as the official build tool so that we can simply ditch 8.0.2. 

Stack downloads a GHC compiler of the specified version and builds a package in an isolated environment, and it doesn't matter what GHC exists in the system---to some extent it is platform-independent. Then, we can avoid hassles related to 8.0.2. Personally, I would like to move up to 8.4 since `Semigroup` is a superclass of `Monoid`. This approach might be too radical, though.

**Comment by andreasabel** [comment]
> use Stack as the official build tool so that we can simply ditch 8.0.2.

I am not using stack and other container systems habitually since they eat up my diskspace so quickly.  The unit of consumption here seems to be GB; if I had a 5000 GB hard drive, I would not care, but with my current capacity it is too much clean-up and maintenence work.  Thus, I am hesistent to commit to stack yet.

Also, some of our clients use the haskell packaged by linux distros, and we want to support the ghc version of the LTS releases.

**Comment by nad** [comment]
```
$ du -sh ghc-* | sort -h
402M    ghc-6.10.4
581M    ghc-6.12.3
601M    ghc-7.0.4
667M    ghc-7.4.2
681M    ghc-7.6.3
748M    ghc-7.8.4
832M    ghc-7.10.3
986M    ghc-8.0.2
1,1G    ghc-8.2.2
1,4G    ghc-8.4.4
1,4G    ghc-8.6.5
```

**Comment by andreasabel** [comment]
Looks like this issue was fixed by @L-TChen  by his work on #4177.
